### PR TITLE
2.6.4 development (address deprecated function and payment fee compatibility ext extension)

### PR DESCRIPTION
**Pull request by Xaxxis**
- Address Magento action deprecated class
- Add payment fee items on payment request, if payment fee data available on Magento order
- Override pending page using finish page instead, so user can get the actual status transaction in frontend after payment
- Remove variable type for backward compatibility

### CODE REVIEW

File: Controller/Payment/Action.php
Comment (rizdaprasetya) [inline Controller/Payment/Action.php]: Things to note for backlogs/future refactor:
`Data` is not so descriptive (both for class name & variable name), what data is it exactly? ConfigurationData? better rename into something that can be easily understand when read.

File: Controller/Payment/Action.php
Comment (rizdaprasetya) [inline Controller/Payment/Action.php]: `$param` is not so descriptive, what is this parameter exactly? I guess a magento order instance? maybe rename to `$order` or `$magentoOrder`

`$code` what code is it exactly? I guess $paymentCode or $paymentExtensionCode

File: Helper/Utils.php
Comment (rizdaprasetya) [inline Helper/Utils.php]: what's the goal of this function? I guess it's to replace items name so that it become safe/valid value for Midtrans charge API param?, hmm then at least include that explanation on the DocString/DocBlock/DocComment. Because the function name itself is not so descriptive, I don't know what's its for.

File: Helper/Utils.php
Comment (rizdaprasetya) [inline Helper/Utils.php]: hmm function name seems to be not so descriptive, I guess the purpose of this function is to "trim" excess string length, so that max length is 20 char.

Also I see it usually used together with previous function:
```
$this->repString($this->getName($newItem->getName()))
```

Maybe merge the 2 function, so that when calling from Controller/Model you just call:
```
$this->sanitizeItemName($newItem->getName())
```

Then internally the `sanitizeItemName` utils function can call `trimLength` & `sanitizeString` internal function.

File: README.md
Comment (rizdaprasetya) [inline README.md]: Fix wording:
```
In case you need it, payment fee can optionally be added by using additional (3rd party) extension: [Mageprince Magento2 PaymentFee](https://github.com/mageprince/magento2-paymentfee/) extension. It allows adding extra charges for specific payment methods and displays them on the cart page, checkout page, invoice, and credit memo.

All the fee calculation will be handled by that extension. Midtrans extension will then take the produced PaymentFee value (from Magento order object), and parse it into additional item object for Midtrans API param.
```

File: README.md
Comment (rizdaprasetya) [inline README.md]: Should be:
> Configuring Custom Payment Fee

File: README.md
Comment (rizdaprasetya) [inline README.md:179]: Likely we will also get asked on tutorial/example on how to use that plugin to add fee, so it's better to show write the steps/screenshots. And put it into a "click to expand" section (like [this](https://github.com/veritrans/SNAP-Woocommerce#customizing-snap-api-parameters))

**Review by rizdaprasetya — COMMENTED** [review]
Actually it's overall good 👍 , ok to merge & release. Just some minor feedback, I'll let you decide to address the review in this PR or for future enhancement. Thanks @Xaxxis

File: Controller/Payment/Action.php
Comment (rizdaprasetya) [reply Controller/Payment/Action.php]: hmm or maybe that is the standard/best practice class name from the Magento extension template? In that case then the `Data` class name is fine.

File: README.md
Comment (Xaxxis) [reply README.md:179]: Updated, please help to review. Regarding the steps of installation and configuration that plugins. Mostly I put their GitHub readme. So if any update from their side, we don't need to update our readme

File: README.md
Comment (Xaxxis) [reply README.md]: Done updated

File: README.md
Comment (Xaxxis) [reply README.md]: done updated

File: Controller/Payment/Action.php
Comment (Xaxxis) [reply Controller/Payment/Action.php]: @rizdaprasetya param it can be `Midtrans order/trx-id` or `Magento order object`, that it. Agree it's not descriptive. Any idea? can I used $midtransOrderIdorMagentoOrder or put the descriptive on PHP docs.

Yes the `$code` it should be `$paymentCode` sure will change this

File: Controller/Payment/Action.php
Comment (Xaxxis) [reply Controller/Payment/Action.php]: done updated and tested, changed class name data to more descriptive. Kindly help check

File: Controller/Payment/Action.php
Comment (Xaxxis) [reply Controller/Payment/Action.php]: Done, updated the PHP docs

File: Helper/Utils.php
Comment (Xaxxis) [reply Helper/Utils.php]: Done, refactored to one function sanitizeItemName

File: README.md
Comment (rizdaprasetya) [inline README.md]: Fix grammar & wording to
```
You can install from [Magento Marketplace](https://marketplace.magento.com/prince-magento2-paymentfee.html) or follow [the manual installation step](https://github.com/mageprince/magento2-paymentfee/#installation-instruction).
```

File: README.md
Comment (rizdaprasetya) [inline README.md]: Fix to:
```
This is just an example configuration that works at the time of this writing, the extension may change in the future. For further & most up to date details of Mageprince PaymentFee extension configuration, you can check  [the official extension documentation](https://github.com/mageprince/magento2-paymentfee/#configuration).
```

File: README.md
Comment (rizdaprasetya) [inline README.md]: Fix sub heading to:
> Example on how to use the PaymentFee Extension

**Review by rizdaprasetya — APPROVED** [review]
Code looks good now, cool thanks @Xaxxis 👍  Basically this PR is approved. Just minor docs/readme feedback, after updating it you can merge & release. Thanks!

**Pull request merged**
